### DISCUSSION TITLE
Coming from MATLAB

### DISCUSSION THREAD

**Discussion by tgkolda** [sandialabs/pyttb#346] [General]
I don't necessarily want to make these as issues just yet, but here just have a discussion of things that might enhance the documentation for folks coming from MATLAB.

* **Python arrays are 0-indexed** - It might be helpful to mention this in a prominent spot!

**Comment by ntjohnson1** [comment]
I think including a reference to the numpy discussion for general differences in numpy and python probably will help cover a lot of the general details not specific to pyttb. https://numpy.org/doc/2.0/user/numpy-for-matlab-users.html

**Reply by tgkolda** [reply]
I would vote for the main points (like 0-indexing) being included in your guide, with just a link to this reference.

**Reply by ntjohnson1** [reply]
Easy enough. I opened a draft [PR](https://github.com/sandialabs/pyttb/pull/352) with [that change](https://pyttb--352.org.readthedocs.build/en/352/matlab/common.html#python-conventions). We can either keep adding to that list or merge this in then add things later as follow up.

At some point I'd like to refactor our whole MATLAB section so that its a jupyter notebook we can test to confirm it doesn't go stale, but that hasn't been high enough priority to put into an issue yet. If there ends up being a lot of confusion or refactor required then a minor overhaul might make sense.

**Comment by ntjohnson1** [comment]
For tracking purposes linking to the other discussion that ended up being about matlab getting started documentation. New addition to the list reference counting in python.

https://github.com/sandialabs/pyttb/discussions/345#discussioncomment-11450172

**Comment by ntjohnson1** [comment]
Everything discussed this far should be merged now. There was a [note](https://github.com/sandialabs/pyttb/pull/352#issuecomment-2540402725) after the PR for additional considerations.

Leaving discussion open for new things discovered.
